app/services/katello/repository_type.rb

```diff
@@ -103,5 +120,36 @@ def label
         self.model_class::CONTENT_TYPE
       end
     end
+
+    class GenericContentType < ContentType
+      attr_accessor :pulp3_api, :pulp3_model, :content_type
+
+      def initialize(options)
+        self.model_class = options[:model_class]
+        self.priority = options[:priority] || 99
+        self.pulp3_service_class = options[:pulp3_service_class]
+        self.index = options[:index].nil? ? true : options[:index]
+        self.index_on_pulp3 = options[:index_on_pulp3].nil? ? true : options[:index_on_pulp3]
+        self.uploadable = options[:uploadable] || false
+        self.removable = options[:removable] || false
+        self.pulp3_api = options[:pulp3_api]
+        self.pulp3_model = options[:pulp3_model]
+        self.content_type = options[:content_type]
+      end
+
+      def label
+        self.model_class::CONTENT_TYPE
+      end
+    end
+
+    class RepositoryRemoteOptions
```

Comment:
`RepositoryRemoteOptions` confused me a little because usually we keep a prefix like `Repository` for joins tables (like `RepositoryErratum`).  Also the class should be singular since it represents a single remote option.  Maybe `GenericRemoteOption`?